### PR TITLE
More informative error messages

### PR DESCRIPTION
**Pull request by AgilentGCMS**
## Types of change(s)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Trivial change (affects only documentation or cleanup)
- [ ] Refactor (no functional changes, no api changes)

## Checklist
- [X] Tested this change with a run of GEOSgcm
- [ ] Ran the Unit Tests (`make tests`)

## Description
* In `base/MAPL_CFIO.F90`, added `Iam` to a print statement so that when a read fails we know which routine failed.
* In `gridcomps/ExtData2G/ExtDataConfig.F90`, print out the name of the duplicate collection that causes model to fail.

### CODE REVIEW

File: gridcomps/ExtData2G/ExtDataConfig.F90 [hunk 1]
```diff
@@ -68,6 +68,7 @@ recursive subroutine new_ExtDataConfig_from_yaml(ext_config,config_file,current_
       logical :: file_found
       logical :: is_right_type
       character(len=:), allocatable :: sub_configs(:)
+      character(len=ESMF_MAXSTR) :: error_message
```
Comment (tclune) [inline gridcomps/ExtData2G/ExtDataConfig.F90:71]:
```suggestion
```
See fix below.  Undesirable to fill variable even when no error exists.

File: gridcomps/ExtData2G/ExtDataConfig.F90 [hunk 2]
```diff
@@ -107,7 +108,8 @@ recursive subroutine new_ExtDataConfig_from_yaml(ext_config,config_file,current_
          do while (ESMF_HConfigIterLoop(hconfigIter,hconfigIterBegin,hconfigIterEnd))
             hconfig_key = ESMF_HConfigAsStringMapKey(hconfigIter,_RC)
             temp_ds => ext_config%file_stream_map%at(hconfig_key)
-            _ASSERT(.not.associated(temp_ds),"defined duplicate named collection")
+            write(error_message, '("defined duplicate named collection ", a)') trim(hconfig_key)
+            _ASSERT(.not.associated(temp_ds), error_message)
```
Comment (tclune) [inline gridcomps/ExtData2G/ExtDataConfig.F90:112]:
```suggestion
           _ASSERT(.not.associated(temp_ds),"defined duplicate named collection " // trim(hconfig_key))
```

**Review by tclune — DISMISSED** [review]

**Review by tclune — DISMISSED** [review]
Subject to passing CI ...

**Comment by tclune** [comment]
@AgilentGCMS Please add an entry in CHANGELOG.md in the top directory.

**Comment by tclune** [comment]
Looks like something went very wrong with the CI. (Unrelated to your change.)  Will discuss with @mathomp4 tomorrow.

**Comment by mathomp4** [comment]
Those are weird errors...

**Comment by mathomp4** [comment]
Okay. I think it was just an old circleci config. I merged in the latest develop and it's happy.

@AgilentGCMS Can you pull and update the CHANGELOG with an entry?

**Review by 818Nawaf — APPROVED** [review]

**Pull request merged**